airbyte-integrations/connectors/source-s3/integration_tests/spec.json

```diff
@@ -316,8 +316,10 @@
       },
       "endpoint": {
         "title": "Endpoint",
-        "description": "Endpoint to an S3 compatible service. Leave empty to use AWS.",
+        "description": "Endpoint to an S3 compatible service. Leave empty to use AWS. The custom endpoint must be secure, but the 'https' prefix is not required.",
         "default": "",
+        "examples": ["my-s3-endpoint.com", "https://my-s3-endpoint.com"],
+        "pattern": "^(?!http://).*$",
```

Comment:
I don't really get why we allow `http` endpoint but raise a validation error if `endpoint.startswith("http://")`
I think it's fine to allow them on OSS but not on Cloud.
To enforce https on Cloud I suggest to use the `is_cloud_environment` function of the CDK.
You can check a PR for a similar use case [here](https://github.com/airbytehq/airbyte/pull/30984/files)

Reply:
@alafanechere The regex pattern is designed to reject `http://` prefixes. Using `is_cloud_environment` won't affect the `check_connector_https_url_only` test since the pattern and check are inside `.py` file.